### DISCUSSION TITLE
Are MeshShape's Static Only?

### DISCUSSION THREAD

**Discussion by DennisSmolek** [jrouwe/JoltPhysics.js#142] [Q&A]
~~I haven't been able to find it in the docs~~ but I did see someone elses implementation that also limited the MeshShape to static. 
Edit: [Found here: ](https://jrouwe.github.io/JoltPhysics/class_mesh_shape.html)

I tried creating them as dynamic and it acts very strange... It's hard to pin down but something isn't right and the object never moves nor renders. 

But if I switch the shape to be static everything works correctly. 

I saw some other issues where it was mentioned as best practices but I didn't know if was a rule.

**Comment by jrouwe** [comment]
I elaborated a bit in the docs here: https://jrouwe.github.io/JoltPhysics/#dynamic-mesh-shapes

I would suggest building a debug build and looking in the console log for any asserts that trigger. Recording a movie of the behavior you see may also help me identify what the issue is.

**Reply by DennisSmolek** [reply]
That's a good idea, I'll try the debug build now

**Comment by DennisSmolek** [comment]
I did a debug build and noticed errors related to me not setting the mass/inertia so I did that and it cleared any errors in the WASM debug build. 

Here's what happens when I set the body to static:
![image](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/861a5b93-e9d2-43bc-8f22-2dab1884ca70)

The shapes all exist, gravity works etc. 

Here's only the `motionType` changing to dynamic:
![image](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/ae821f8f-f14b-4de8-9ffe-790e7bb3a07a)
![image](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/7dd7ed72-58f5-4970-a768-44007f26bb25)

The trimesh shape ignores the position X and Z values and has a negative Y offset to where it appears to fall below the "floor" shape. What's interesting is it does collide with it as you can watch it "fall". 

If I change the shape type to box, sphere, or convex, it works totally as expected:
![image](https://github.com/jrouwe/JoltPhysics.js/assets/1397052/5847bb6e-d369-47ab-8135-222a7a13871f)

**Reply by DennisSmolek** [reply]
Ok it's something to do with local space/relative positioning.

Normally when I call `GetPosition` the values come back in worldSpace, but when I use a dynamic trimesh they are coming back in local space

**Reply by jrouwe** [reply]
`Body::GetPosition` should always return a position in world space, I'm not sure what you mean with 'local space' (local to what?). I tried changing the C++ sample DynamicMeshTest to use a mesh that's offset from the center of mass and that seems to work fine (you get a funny sim if the mesh is not centered around the COM but it does work).

Do you perhaps have a repro? (maybe you can modify one of the examples)
